### PR TITLE
Improve layout styles with flexbox

### PR DESCRIPTION
**Pull request by rtlong**
Also, yank out Foundation

### CODE REVIEW

**Comment by rtlong** [comment]
## Before

screen shot 2015-07-06 at 01 08 43
screen shot 2015-07-06 at 01 09 46

## After

screen shot 2015-07-06 at 01 10 24
screen shot 2015-07-06 at 01 10 14

## Notes
- Album art is "zoomed" to fit square box
- Scroll bar is just inside tracklist section and is styled
- in Player, text element is vertically centered on the play/pause button
- play progress and timer components were separated
  - play progress bar is colored using two colors which are visible
  - updates to the play progress bar are smoothed using the CSS `transition` property
    - this includes when clicking on it, which I think adds a nice touch
- tracks are centered horizontally
- tracks are centered vertically when list is short enough (maybe this isn't what we want, idk, but I feel it helps to emphasize the tracks being the focal point of the UX so far)
- track hover opacity change is also transitioned smoothly using `transition`
- track title and artist names are length-limited so they do not wrap to more than 1 line with an ellipsis should that limit take effect

**Comment by rtlong** [comment]
Also, please try to ignore the restart stuff here, I merged that in already: #9

File: lib/components/app.js [hunk 1]
```diff
@@ -2,21 +2,69 @@ import React from 'react'
...
+        alignContent: 'center',
+        padding: '0.5em',
+      },
+      tracklist: {
+        flex: '0 0 100%',
+      },
+      footer: {
+        flex: '0 0 auto',
+      },
+    }
```
Comment (namick) [inline lib/components/app.js:52]: I like all the styles in a single object.  Big improvement over dealing with several objects or methods for styles.

File: lib/components/play-pause-button.js [hunk 2]
```diff
@@ -16,10 +17,10 @@ class PlayPauseButton extends React.Component {
   render() {
     var style = {
       padding: '0.5em',
-      marginBottom: 0,
       backgroundColor: '#C36351',
-      borderRadius: '10',
+      borderRadius: '1em',
```
Comment (namick) [inline lib/components/play-pause-button.js:21]: I do think we could turn the rounded corner down a bunch.  maybe to 3 or 4 px - or 0.25em

File: lib/components/playback-time.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    var seconds = Math.floor(divisorForSeconds)
+    if (seconds < 10) { seconds = '0' + seconds }
+
+    return {
+      h: hours,
+      m: minutes,
+      s: seconds,
+    }
+  }
+}
```
Comment (namick) [inline lib/components/playback-time.js:46]: Nice, these needed to be broken off into their own components.

File: static/style.css [hunk 4]
```diff
@@ -1,8 +1,75 @@
...
+}
+
+::-webkit-scrollbar-track {
+    opacity: 0;
+}
+
+::-webkit-scrollbar-thumb {
+    border-radius: 8px;
+    background: rgb(50,52,56);
+}
```
Comment (namick) [inline static/style.css:75]: Yay, no vendored styles.

**Comment by namick** [comment]
All good.  Big win.

**Pull request merged**